### PR TITLE
LPS-155839: Don't display web content template titles in escaped format

### PR DESCRIPTION
**Pull request by noornajj**
@jakesliwka

Forwarded from https://github.com/noornajj/liferay-portal/pull/12

> https://issues.liferay.com/browse/LPS-155839
>  
>  The issue is caused when the user creates a Template in Web Content. If the user used certain characters (such as &, ", ', < >, etc), the title of the Template would display correctly at the top, but once the user navigates back to the Templates page to view all of the templates, the titles would be displayed in escaped format. This bug was tracked to JournalDDMTemplateVerticalCard.java where the getName function for the Templates was being used. It was returning the HtmlUtil.escape format of the title, where it should just be returning the original title. This fixes that return statement and allows for the template titles to be displayed properly with any characters.

Test failures in previous PR were confirmed as not related.

### CODE REVIEW

**Comment by noornajj** [comment]
Thank you @ericyanLr 
@jakesliwka please see Eric's analysis

**Comment by jakesliwka** [comment]
@ericyanLr 

After investigating further, it was found that:
**Roles** and **Templates** both return with escaped name formats while utilizing **HtmlUtil.escape**, whereas **Sites** returns a correctly displayed name while also utilizing **HtmlUtil.escape** 

Because:

At first, inside of **SiteVerticalCard.java**, where the getTitle() is being called, it uses getDescriptiveName which is wrapped in the HtmlUtil.escape, which returns the escaped name. Since the site name was still displaying correctly, even though it is returning the name as escaped, it is clear that the solution must come from the .jsp file that calls it. 

So, after viewing **view_sites.jsp** and noting its differences from **view_ddm_templates.jsp** and **role_columns.jsp**, the main one was that both **view_ddm_templates.jsp** and **role_columns.jsp** get their title values from this line:

`verticalCard="<%= new RoleVerticalCard(role, renderRequest, searchContainer.getRowChecker()) %>"`
wrapped inside of a
`<clay:vertical-card/>`
which is **automatically** generating a vertical card using an object, whereas **view_sites.jsp** uses this:

`<%= siteVerticalCard.getTitle() %>`
which is **manually** creating the card, specifically its **title**, using a string inside of a JavaScript evaluator, thus requiring that it is escaped to prevent JavaScript execution.

So, the main differences show that the **view_ddm_templates.jsp** and **select_roles.jsp** utilize automatic generation of these vertical cards, whereas the **view_sites.jsp** manually generate parts of the card.

 
To verify this, I kept my original solution of removing the HtmlUtil.escape from anything relating to the templates and gave a template this name:

_<script>alert("test");</script>_

I also removed the HtmlUtil.escape from the getDescriptiveName method that the sites were using to verify that my solution wouldn't work on that, and created a site with the same name.

Upon viewing, the **template** name showed up normally, displaying "<script>alert("test");</script>" as the title and not executing anything. However, when I viewed my **sites** with the new name, the alert window came up on my screen, showing that the script was ran.
This shows that the sites _require_ the name to be escaped because of how it is formatted in the **view_sites.jsp**, whereas with the templates, removing the HtmlUtil.escape from the return statement does not have an impact on actually executing the JavaScript, and only fixes the escaped name issue.
I also tried it with **roles**, since it follows a similar format in **role_columns.jsp** than that of **view_ddm_templates** and created a site role with the name "<script>alert("test");</script>", only to see that it has the same outcome of displaying the correct names and not executing any JavaScript.


So, to summarize, I believe that my original solution of removing the **HtmlUtil.escape** from the return statements inside of anything **template** related, with the new addition of anything **role** related, could be the correct solution for reasons stated above. I apologize if this explanation was unclear, and I would be happy to clarify anything. 
The next steps for me are to verify that the updated return statements aren't being used in any files inside of a JavaScript evaluator, as well as check for other modules using the same format and to update those as well.

**Comment by ericyanLr** [comment]
Hi @jakesliwka ,

Thanks for sharing your findings, your explanation of the two different behaviors makes sense to me.

Also, your next steps sounds good to me.
